src/components/AddCarForm/AddCarForm.tsx

```diff
@@ -0,0 +1,473 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { v4 as uuidv4 } from 'uuid';
+import { useEffect, useState } from 'react';
+import SuccessCreationMessage from '../SuccessCreationMessage/SuccessCreationMessage';
+import { GET_CARS } from '../../api/graphql/query/cars';
+
+export type IFormInput = {
+  price: number | null;
+  odometer: number | null;
+  vin: string;
+  title: string;
+  damage: string;
+  year: Date;
+  description: string;
+  saleDate: Date;
+  condition: string;
+  brand: number | null;
+  model: number | null;
+  color: number | null;
+  state: number | null;
+  city: number | null;
+};
+
+const defaultValuesObject = {
+  title: '',
+  odometer: null,
+  vin: '',
+  brand: null,
+  model: null,
+  saleDate: new Date(),
+  price: null,
+  condition: '',
+  damage: '',
+  year: new Date(),
+  color: null,
+  state: null,
+  city: null,
+  description: '',
+};
+
+const schema = yup.object().shape({
+  title: yup.string().required(),
+  odometer: yup
+    .number()
+    .required()
+    .positive()
+    .typeError('Please insert a valid ODOmeter'),
+  vin: yup.string().required(),
+  price: yup
+    .number()
+    .required()
+    .positive()
+    .typeError('Please insert a valid Price'),
+  damage: yup.string().required(),
+  description: yup.string().required(),
+  saleDate: yup.date().required().typeError('Sale Date is required'),
+  year: yup.date().required().typeError('Vehicle Year is required'),
+  condition: yup.string().required().typeError('Please Select a condition'),
+  brand: yup.number().required().typeError('Please Select a brand'),
+  model: yup.number().required().typeError('Please Select a Model'),
+  color: yup.number().required().typeError('Please Select a Color'),
+  state: yup.number().required().typeError('Please Select a Color'),
+  city: yup.number().required().typeError('Please Select a Color'),
+});
+
+const AddCarForm = () => {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    control,
+    reset,
+  } = useForm<IFormInput>({
+    resolver: yupResolver(schema),
+    defaultValues: defaultValuesObject,
+  });
+
+  const [isSuccessCreation, setIsSuccesCreation] = useState<boolean>(false);
+
+  const { title, odometer, vin, price, damage, description } = fieldNameValues;
+
+  const { data, error, loading } = useGet_Add_Car_Fields_QueryQuery();
+
+  const [
+    getModelsLazyQuery,
+    { data: lazyModelData, loading: lazyModelLoading },
+  ] = useGet_Add_Car_Fields_QueryLazyQuery();
+
+  const [
+    getCitiesLazyQuery,
+    { data: lazyCitiesData, loading: lazyCitiesLoading },
+  ] = useGet_Add_Car_Fields_QueryLazyQuery();
+
+  const [
+    insertCar,
+    { data: mutationData, loading: mutationLoading, error: mutationError },
+  ] = useMutation_CarsMutation();
+
+  useEffect(() => {
+    if (mutationData) {
+      if (mutationData!.insert_cars!.returning.length > 0) {
+        reset(defaultValuesObject);

```

Comment:
Maybe. It's been a while since I've used apollo but I'm pretty sure all the information about the request is available in the mutation hook result. @stevenheffner probably remembers better